### DISCUSSION TITLE
What parameters should I use to make sure that a text is rendered nicely?

### DISCUSSION THREAD

**Discussion by GasimGasimzada** [Chlumsky/msdf-atlas-gen#45] [Q&A] [answered]
I was able to create a MTSDF font atlas in using this library and render the glyphs to the screen using shaders. I went a bit of a brute force approach and just rendered every glyph in a quad that is bound by `(0.0), (1, 0), (0, 1), (1, 1)`:

![image](https://user-images.githubusercontent.com/807231/180654516-9b377001-9c1b-4fd4-9b68-b90387b27061.png)

This did not give the most perfect result and I see two problems here:

1. The scale of glyphs are incorrect. For example, the letter "l" is too wide and does not look nice
2. The spacing between these glyphs is not correct

What parameters do I need to use to from the Glyph information in order to render the text in a consistent way. I am retrieving glyph bounds in the following way:

```cpp
std::vector<GlyphGeometry> msdfGlyphs;
std::unordered_map<uint32_t, FontGlyph> glyphMap;

for (auto &msdfGlyph : msdfGlyphs) {
  // Engine specific structure for glyphs
  FontGlyph glyph{};

  double top = 0.0, left = 0.0, bottom = 0.0, right = 0.0;
  msdfGlyph.getQuadAtlasBounds(left, bottom, right, top);
  
  // Note: fHeight - bottom/top is used since I need
  // the atlas to be flipped to work properly in Vulkan
  glyph.bounds = glm::vec4(static_cast<float>(left),
                           static_cast<float>(fHeight - bottom),
                           static_cast<float>(right),
                           static_cast<float>(fHeight - top)) /
                 fWidth;

  glyphMap.insert_or_assign(msdfGlyph.getCodepoint(), glyph);
}
```

What values do I need to use from `GlyphGeometry` structure to determine the scale and the spacing between glyphs when rendering text?

**Comment by Chlumsky** [comment, marked as answer]
`getQuadPlaneBounds` and `getAdvance` for spacing.

**Reply by GasimGasimzada** [reply]
Thank you! I finally got it working!
